### PR TITLE
beer-song: Rewrite tests and add working stub solution.

### PR DESCRIPTION
**Pull request by rbasso**
- Add `HINTS.md` explaining that the exercise is about refactoring.
- Use lyrics from `x-common/exercises/beer-song/description.md`.
- Tests all lines, one by one, and then the entire song.
- Remove old tests checking verses.
- Add stub solution passing the tests.
- Patch example solution to work with the new tests.

Related to #301.

### CODE REVIEW

File: exercises/beer-song/src/Example.hs [hunk 1]
```diff
@@ -30,5 +30,5 @@ verse n = unlines
   [ capitalize (stanza n) ++ " on the wall, " ++ stanza n ++ "."
   , action n ++ ", " ++ stanza (pred n) ++ " on the wall." ]
 
-sing :: Int -> Int -> String
-sing from to = unlines $ map verse [from, from - 1 .. to]
+sing :: String
+sing = init $ unlines $ map verse [99, 98 .. 0]
```
Comment (petertseng) [inline exercises/beer-song/src/Example.hs]: we were inconsistent! Sometimes using `init` and `unlines` (https://github.com/exercism/xhaskell/blob/master/exercises/food-chain/src/Example.hs#L18), sometimes using `intercalate "\n"` (https://github.com/exercism/xhaskell/blob/master/exercises/house/src/Example.hs#L6)!

I don't actually care though.

File: exercises/beer-song/src/Example.hs [hunk 2]
```diff
@@ -30,5 +30,5 @@ verse n = unlines
   [ capitalize (stanza n) ++ " on the wall, " ++ stanza n ++ "."
   , action n ++ ", " ++ stanza (pred n) ++ " on the wall." ]
 
-sing :: Int -> Int -> String
-sing from to = unlines $ map verse [from, from - 1 .. to]
+sing :: String
```
Comment (petertseng) [inline exercises/beer-song/src/Example.hs]: should we take the opportunity to change to `song` to match food-chain's https://github.com/exercism/xhaskell/blob/master/exercises/food-chain/src/Example.hs#L17 ?

Comment (rbasso) [inline exercises/beer-song/src/Example.hs]: Done!

Now `house` is the different one, using `rhyme`.

**Comment by petertseng** [comment]
Seems good!

**Pull request merged**